Kas 3794 - search documents route

kobemertens opened the pull request:
Added document indexing

kobemertens committed:
added basic pieces search route

kobemertens committed:
add documentType en confidentialityLevel to index

kobemertens committed:
index ids instead of uris

kobemertens committed:
added document content to index

kobemertens committed:
renamed content to data

sergiofenoll committed:
Merge branch 'development' into KAS-3794

sergiofenoll committed:
feat: add mandatee ids to piece index and index piece access level uris

sergiofenoll committed:
Merge branch 'development' into KAS-3794

ValenberghsSven commented on config/dispatcher/dispatcher.ex:
```diff
@@ -200,6 +200,11 @@ defmodule Dispatcher do
   match "/document-containers/*path", @json_service do
     Proxy.forward conn, path, "http://cache/document-containers/"
   end
+
+  match "/pieces/search/*path", @json_service do
+    Proxy.forward conn, path, "http://search/pieces/search/"
+  end
+
```
there is a search section on line 70-90

ValenberghsSven commented on config/search/config.json:
```diff
@@ -295,6 +295,70 @@
+        "accessLevel": [
+          "http://data.vlaanderen.be/ns/besluitvorming#vertrouwelijkheidsniveau"
+        ],
+        "mandateeIds": [
+          "^http://data.vlaanderen.be/ns/besluitvorming#geagendeerdStuk",
+          "http://mu.semte.ch/vocabularies/ext/heeftBevoegdeVoorAgendapunt",
+          "http://data.vlaanderen.be/ns/mandaat#isBestuurlijkeAliasVan",
+          "http://mu.semte.ch/vocabularies/core/uuid"
+        ],
```
This restricts all documents search results with mandatees to documents of agendaitems.
Is this the intention or do we want to be able to search all documents across the board?
Perhaps we need to not have a minister filter in the documents search route

sergiofenoll replied:
Good point. I also don't think mu-search will allow us as-is to check for branching paths (i.e. gather all mandatees via paths to the agendapoint, via paths to the subcase, via paths to the publication, etc.).

Will double check with Hannes what the intention was.

ValenberghsSven replied:
Unsure of how search works on this. Could we define multiple paths for the same doc?
1 doc can be connected to a subcase (via submission), to a different subcase (linked doc directly), agendaitem (directly on multiple versions), to a decisionmaking-Flow (directly by sync-service), to a publicationFlow (if VIA MR)

sergiofenoll replied:
I think we could have `agendaitemMandateeIds`, `agendaitemLinkedMandateeIds`, `subcaseMandateeIds`, ... and then in the frontend form the query correctly so that the selected mandatees are found in one or more of those fields.

ValenberghsSven replied:
Currently, the search results of docs open in document preview.
We have no way in the frontend to go from that preview to another context (like the agendaitem or subcase doc view) because there can be many possible contexts.

ValenberghsSven replied:
> I think we could have `agendaitemMandateeIds`, `agendaitemLinkedMandateeIds`, `subcaseMandateeIds`, ... and then in the frontend form the query correctly so that the selected mandatees are found in one or more of those fields.

All of those params would end up in the address bar + all the id's.
Pretty sure we are going to hit a limit of address length at some point

sergiofenoll replied:
> > I think we could have `agendaitemMandateeIds`, `agendaitemLinkedMandateeIds`, `subcaseMandateeIds`, ... and then in the frontend form the query correctly so that the selected mandatees are found in one or more of those fields.
> 
> All of those params would end up in the address bar + all the id's. Pretty sure we are going to hit a limit of address length at some point

No I meant that the index will contain those fields, and then the request we send will have those fields specified as a filter (so something like `filter[:terms:agendaitemMandateeIds, subcaseMandateeIds, ...]=aaaa-bbbb-cccc-dddd`. The query param will just remain `ministers` with one flat list. So the user looks for documents linked to e.g. Jambon, no matter through which path.

sergiofenoll committed:
feat: documents' mandatees are determined based on subcases

Before we used the agendaitem, now we use the subcase. Ideally we'd just
use all the linked items, but that makes the filtering quite hard, so we
will just stick to subcases for now.

sergiofenoll committed:
chore: move pieces search config in dispatcher to search section
